pkg/types/vsphere/conversion/installconfig.go

```diff
@@ -44,23 +96,50 @@ func ConvertInstallConfig(config *types.InstallConfig) error {
 			}
 		}
 	}
+}
 
+func fixTechPreviewZonalFailureDomainsScenario(platform *vsphere.Platform, finder *find.Finder) error {
 	// Scenario: Fields are not paths
 	if len(platform.FailureDomains) > 0 {
+		var err error
+
 		for i := range platform.FailureDomains {
-			platform.FailureDomains[i].Topology.ComputeCluster = setComputeClusterPath(platform.FailureDomains[i].Topology.ComputeCluster,
-				platform.FailureDomains[i].Topology.Datacenter)
+			computeCluster := platform.FailureDomains[i].Topology.ComputeCluster
+			datastore := platform.FailureDomains[i].Topology.Datastore
+			folder := platform.FailureDomains[i].Topology.Folder
+			datacenter := platform.FailureDomains[i].Topology.Datacenter
 
-			platform.FailureDomains[i].Topology.Datastore = setDatastorePath(platform.FailureDomains[i].Topology.Datastore,
-				platform.FailureDomains[i].Topology.Datacenter)
+			if !path.IsAbs(computeCluster) {
```

Comment:
For the sake of making hive's use of these funcs cleaner, could we move the abspath check inside?